pkg/api/validation/validation.go

```diff
@@ -383,8 +383,7 @@ func validateVolumeSource(source *api.VolumeSource, fldPath *field.Path) field.E
 	if source.EmptyDir != nil {
 		numVolumes++
 		if !utilfeature.DefaultFeatureGate.Enabled(features.LocalStorageCapacityIsolation) {
-			unsetSizeLimit := resource.Quantity{}
-			if unsetSizeLimit.Cmp(source.EmptyDir.SizeLimit) != 0 {
+			if source.EmptyDir.SizeLimit != nil && source.EmptyDir.SizeLimit.Cmp(resource.Quantity{}) > 0 {
```

Comment:
I am thinking I only need to check sizeLimit != nil. No matter >0, <0 or =0 should not be allowed